supplementary_materials/notebooks/phi3_rag_on_client.ipynb

```diff
@@ -0,0 +1,992 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from operator import itemgetter\n",
+    "\n",
+    "\n",
+    "chain = (\n",
+    "    {\n",
+    "        \"context\": (itemgetter(\"question\") | retriever),\n",
+    "        \"question\": itemgetter(\"question\")\n",
+    "    } | prompt\n",
+    ")\n",
+    "     \n",
+    "print(chain.invoke({\"question\": question}, config=RunnableConfig(metadata={\"top_k\": 1})).to_string())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eabd5651-db4a-423e-9db4-eb157f6418b6",
+   "metadata": {},
+   "source": [
+    "That's it, we are set to run the prompt through our model.\n",
+    "Next we will initialize our OpenVINO optimized Phi-3 model in a pipeline and form the complete chain to produce an answer to our question"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b1ef00cf-82d5-4020-8d7d-6ac41e5e4fa4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Uncomment to install dependencies\n",
+    "# ! pip install optimum[openvino,nncf]\n",
+    "# Phi-3 is not supported yet in the official release of `intel-optimum` so we will need to install from source\n",
+    "# ! pip install git+https://github.com/huggingface/optimum-intel"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ae5a5086-9ed7-45f1-8837-bc1b2d174616",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_community.llms.huggingface_pipeline import HuggingFacePipeline\n",
+    "from optimum.intel import OVModelForCausalLM, OVWeightQuantizationConfig\n",
+    "from functools import wraps\n",
+    "\n",
+    "\n",
+    "model_name = \"microsoft/Phi-3-mini-4k-instruct\"\n",
+    "save_name = model_name.split(\"/\")[-1] + \"_openvino\"\n",
+    "precision = \"f16\"\n",
+    "quantization_config = OVWeightQuantizationConfig(\n",
+    "    bits=4,\n",
+    "    sym=False,\n",
+    "    group_size=128,\n",
+    "    ratio=0.8,\n",
+    ")\n",
+    "device = \"gpu\"\n",
+    "saved = os.path.exists(save_name)\n",
+    "load_kwargs = {\n",
+    "    \"device\": device,\n",
+    "    \"ov_config\": {\n",
+    "        \"PERFORMANCE_HINT\": \"LATENCY\",\n",
+    "        \"INFERENCE_PRECISION_HINT\": precision,\n",
+    "        \"CACHE_DIR\": os.path.join(save_name, \"model_cache\"),  # OpenVINO will use this directory as cache\n",
+    "    },\n",
+    "    \"quantization_config\": quantization_config,\n",
+    "    \"trust_remote_code\": True,\n",
+    "    \"export\": not saved,\n",
+    "}\n",
+    "\n",
+    "ov_llm = HuggingFacePipeline.from_model_id(\n",
+    "    model_id=model_name if not saved else save_name,\n",
+    "    task=\"text-generation\",\n",
+    "    backend=\"openvino\",\n",
+    "    model_kwargs=load_kwargs,\n",
+    ")\n",
+    "\n",
+    "if not saved:\n",
+    "    # For some reason LC passes the model_kwargs to the tokenizer aswell and this can cause issues when saving\n",
```

Comment:
@OpenVINO-dev-contest, do you know something about this issue? Can we try to handle it on LangChain side?